### PR TITLE
feat(file-cache): Endpoints to remove/retry file cache errors

### PR DESCRIPTION
**Pull request by kukukk**

### CODE REVIEW

**Review by burgerni10 — CHANGES_REQUESTED** [review]

File: src/web-server/routes/index.js [hunk 1]
```diff
@@ -39,4 +40,10 @@ router.put('/history-queries/:id/order', historyQueryController.orderHistoryQuer
 router.delete('/history-queries/:id', historyQueryController.deleteHistoryQuery)
 router.get('/history-queries/:id/status', historyQueryController.getStatus)
 
+router.get('/file-cache/searchErrors', fileCacheController.searchErrors)
+router.delete('/file-cache/removeErrors', fileCacheController.removeErrors)
+router.post('/file-cache/retryErrors', fileCacheController.retryErrors)
+router.delete('/file-cache/removeAllErrors', fileCacheController.removeAll)
+router.post('/file-cache/retryAllErrors', fileCacheController.retryAll)
```
[the remark this answers is not in the document]
Comment (kukukk) [reply src/web-server/routes/index.js:47]: retry-all and remove-all should be valid for a single north?
At the moment the implementation iterates over every active north.

File: src/service/utils.js [hunk 2]
```diff
@@ -150,6 +150,17 @@ const filesExists = async (filePath) => {
   return true
 }
 
+/**
+ * Async filter implementation
+ * @param {array} array - The array to be filtered
+ * @param {function} predicate - The async filter function
+ * @returns {Promise<array>} - The filtered result
+ */
+const asyncFilter = async (array, predicate) => {
```
[the remark this answers is not in the document]
Comment (kukukk) [reply src/service/utils.js:159]: I put it in utils.js to make generally available for the whole project as general async filter implementation. In this case the argument naming should be something general.

File: src/web-server/routes/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (burgerni10) [reply src/web-server/routes/index.js:47]: Yes, it should remove all cached for in this specific north

File: src/service/utils.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (burgerni10) [reply src/service/utils.js:159]: If this can be anything, maybe it's better to change the filename variable l 161 into something else, like element ?

**Review by burgerni10 — APPROVED** [review]

**Pull request merged**